programs/associated-token-account/Create.go

```diff
@@ -191,10 +201,20 @@ func (inst *Create) EncodeToTree(parent treeout.Branches) {
 }
 
 func (inst Create) MarshalWithEncoder(encoder *bin.Encoder) error {
-	return encoder.WriteBytes([]byte{}, false)
+	if inst.Idempotent {
+		if err := encoder.Encode(inst.Idempotent); err != nil {
+			return err
+		}
+	}
+	return nil
```

Comment:
This is the real bit of the change. If the `Idempotent` flag is set, we write it to the encoder. The boolean `true` gets encoded to a `1`, which is what we want in the data array